R/swagger.R

```diff
@@ -191,5 +346,69 @@ removeNaOrNulls <- function(x) {
   ret <- lapply(x, removeNaOrNulls)
   class(ret) <- class(x)
 
+  # Put example back in
+  if (saveExample) {
+    ret$example <- savedExample
+  }
+
   ret
 }
+
+#' For openapi definition
+#' @noRd
+priorizeProperty <- function(...) {
+  l <- list(...)
+  if (length(l) > 0L) {
+    isnullordefault <- vapply(l, function(x) {isNaOrNull(x) || isTRUE(attributes(x)$default)}, logical(1))
+    # return the position of the first FALSE value or position 1 if all values are TRUE
+    return(l[[which.min(isnullordefault)]])
+  }
+  return()
+}
+
+#' Check if x is JSON serializable
+#' @noRd
+isJSONserializable <- function(x) {
+  testJSONserializable <- TRUE
+  tryCatch(toJSON(x),
+           error = function(cond) {
+             # Do we need to test for specific errors?
+             testJSONserializable <<- FALSE}
+  )
+  testJSONserializable
+}
+
+#' Extract metadata on args of plumberExpression
+#' @noRd
+getArgsMetadata <- function(plumberExpression){
+  #return same format as getTypedParams or params?
+  if (!is.function(plumberExpression)) plumberExpression <- eval(plumberExpression)
+  args <- formals(plumberExpression)
+  lapply(args[!names(args) %in% c("...", "res", "req")], function(arg) {
+    required <- identical(arg, formals(function(x){})$x)
+    if (is.call(arg) || is.name(arg)) {
```

Comment:
Maybe we should always be always be `eval()`ing

Reply:
Could you detail your reasoning, I'm confused since your stated otherwise above. Anything we need to aware of?

Reply:
Mainly just to make sure we're not missing a case of some un-evaluated object (I can't think of one at the moment, though).

It also makes it a bit more clear what the intention is (that is, give the evaluated argument value)